src/dev-app/dev-app/dev-app-theme.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Inject, Injectable} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+
+const isDarkThemeKey = 'ANGULAR_COMPONENTS_DEV_APP_DARK_THEME';
+
+@Injectable({providedIn: 'root'})
+export class DevAppTheme {
+  readonly darkThemeClass = 'demo-unicorn-dark-theme';
+  private _isDark = false;
+
+  constructor(@Inject(DOCUMENT) private _document: Document) {
+    let isDark: string | null;
+    try {
+      isDark = localStorage.getItem(isDarkThemeKey);
+      if (isDark != null) {
+        // We avoid calling the setter and apply the themes directly here.
+        // This avoids writing the same value, that we just read, back to localStorage.
+        this._isDark = isDark === 'true';
+        this.updateThemeClass(this._isDark);
+      }
+    } catch (error) {
+      console.error(`Failed to read ${isDarkThemeKey} from localStorage: `, error);
+    }
+  }
+
+  get isDark(): boolean {
+    return this._isDark;
+  }
+
+  set isDark(value: boolean) {
+    // Don't accept null as there is no concept of clearing the theme. It has to be light or dark.
+    // Noop if the value is the same as is already set.
+    if (value != null && value !== this._isDark) {
```

Comment:
In what cases can the value be `null` here? We typed it as `boolean` and have strict null checks enabled, so I think we can remove this.